doc/module-docs/cc_raspberry_pi/data.yaml

```diff
@@ -2,10 +2,11 @@ cc_raspberry_pi:
   description: |
     This module handles ARM interface configuration for Raspberry Pi.
 
-    It also handles Raspberry Pi Connect installation and enablement.
-    Raspberry Pi Connect service will be installed and enabled to auto start on boot.
+    It also handles Raspberry Pi USB Gadget setup.
+    The USB Gadget mode will be enabled to auto start on boot.
+    Make sure you read the documentation on it carefully before enabling it.
 
-    This only works on Raspberry Pi OS (bookworm and later).
+    This only works on Raspberry Pi OS (trixie and later).
```

Comment:
Given that cloud-init 25.2 had initial support for this config module in raspberry-pi-os, we typically need to look at migration path for users which may be providing old user-data cloud-config keys `enable_rpi_connect`  which look like they will now be invalid once we land this PR. A couple of questions for our context:
* Do you expect Raspberry Pi consumers to know whether they are launching bookworm-based images vs trixie and provide the appropriate specialized user-data for both?
* Do you expect cloud-init version 25.4 to ever make it back into Raspberry Pi OS bookworm-based images via package upgrades where you may have scenarios where the old-way is still supported?

Reply:
I understand the concern, but in this case it shouldn’t be an issue. Cloud-init hasn’t yet been shipped on Raspberry Pi OS, so there are no existing users depending on the old behavior.
Going forward, the focus will be on Raspberry Pi OS 6.0 (based on trixie). Once these PRs are merged, that release will include cloud-init and replace firstRun.sh. We don’t plan to publish cloud-init for bookworm or older releases, so migration considerations only apply starting with trixie.